Sources/POSIX/Error.swift

```diff
@@ -43,6 +43,13 @@ import var libc.ERANGE
 extension SystemError: CustomStringConvertible {
     public var description: String {
         func strerror(_ errno: Int32) -> String {
+#if os(Linux)
+            var buf = [Int8](repeating: 0, count: 128)
+            // The GNU flavor returns a char* instead of an int.
+            // It may point to the buf passed in or some other immutable string
+            let err = libc.strerror_r(errno, &buf, buf.count)
```

Comment:
We are not using `err` to form the string anyway so why do we need to special case this? Can you explain what exactly is the problem (possibly using a test case)?

Reply:
I'm using err for the string in this ifdef instead of buf now below. 

I'm attempting to get the clang importer to define _GNU_SOURCE so we can get all the POSIX/XOPEN/C/GNU/BSD extended C functions we are missing out today in glibc.modulemap. 

The only issue with that change is that that strerror_r is the one function out of all libc that glibc emits with a different signature when that feature macro is enabled, so it was causing a build break in swiftpm that is change fixes. We are discussing potential other work arounds in that PR though that don't break this function's signature.

Reply:
The type of `buf` still remains same, right? Why do we want to use `err` instead of `buf`?

Reply:
In GNU's version, they may or may not use the buffer passed in for the string they return from the function. Sometimes, depending on the error, they will return an immutable string for the error string instead. 

I'm going to start a thread on swift-dev to discuss the change to glibc. Per @gparker42 and @jrose-apple, it might require an evolution proposal. There is a chance we can come up with a solution where we don't need to break this function.